test_spack.py

```diff
@@ -0,0 +1,214 @@
+        'git clone git@github.com:MeteoSwiss-APN/cosmo.git \
+            && cd cosmo \
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=gpu +cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html 
+        'git clone git@github.com:MeteoSwiss-APN/cosmo.git \
+            && cd cosmo \
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=cpu ~cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore ^eccodes +aec +build_shared_libs',
+    },
+    'cosmo-dycore' : {},
+    'cosmo-eccodes-definitions' : {},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {},
+    'cuda' : {},
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {},
+    'dyicon_benchmarks' : {},
+    'ecbuild' : {},
+    'eccodes' : {},
+    'eckit' : {},
+    'gridtools' : {},
+    'icon' : {
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_cpu \
+        #     && cd pgi_cpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=cpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_gpu \
+        #     && cd pgi_gpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=gpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'icondusk-e2e' : {},
+    'icontools' : {},
+    'int2lm' : {
+        'spack install int2lm@c2sm_master%pgi', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'spack install int2lm@org_master%pgi pollen=False', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'libgrib1' : {},
+    'mpich' : {},
+    'oasis' : {},
+    'omnicompiler' : {},
+    'omni-xmod-pool' : {},
+    'openmpi' : {},
+    'serialbox' : {},
+    'xcodeml-tools' : {},
+    'zlib_ng' : {},
+}
+
+def UseCases(packages : set):
+    return {case for p in packages for case in use_cases[p]}
+
+# Maps packages to Set[packages it depends on].
+dependencies = {
+    'atlas' : {'ecbuild', 'eckit'},
+    'atlas_utilities' : {'atlas', 'eckit'},
+    'claw' : {},
+    'cosmo' : {'cuda', 'serialbox', 'libgrib1', 'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'omni-xmod-pool', 'claw', 'zlib_ng'},
+    'cosmo-dycore' : {'gridtools', 'serialbox', 'cuda'},
+    'cosmo-eccodes-definitions' : {'eccodes'},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {'cosmo-grib-api'},
+    'cuda' : {},
+    'dawn' : {},
```

Comment:
dawn depends on cuda

Reply:
Where is this dependency defined?
Because there's no `depends_on(cuda)` in the [dawn package](https://github.com/C2SM/spack-c2sm/blob/PR_testing/packages/dawn/package.py). Or is it missing in there?

Reply:
true as well. dawn repo depends on cuda via gtclang (which is included in the repo). But the spack package refers to only dawn, which does not require cuda